plugins/module_utils/module_helper.py

```diff
@@ -11,6 +12,21 @@
 from ansible.module_utils.basic import AnsibleModule
 
 
+class ModuleHelperException(Exception):
+    @staticmethod
+    def _get_remove(key, kwargs):
+        if key in kwargs:
+            result = kwargs[key]
+            del kwargs[key]
+            return result
+        return None
+
+    def __init__(self, *args, **kwargs):
+        self.msg = self._get_remove('msg', kwargs) or "Module failed with exception: {0}".format(str(self))
```

Comment:
Does `str(self)` does something reasonable before `super()` is called?

Reply:
TBH I no longer recall why I put ``str()`` in there, it does not make much of a sense to me now. Will remove it.
```suggestion
        self.msg = self._get_remove('msg', kwargs) or "Module failed with exception: {0}".format(self)
```